spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -230,8 +266,14 @@ public void testRewriteDataFilesWithFilterOnPartitionTable() {
     assertEquals(
         "Action should rewrite 5 data files from single matching partition"
             + "(containing c2 = bar) and add 1 data files",
-        ImmutableList.of(row(5, 1)),
-        output);
+        row(5, 1),
+        Arrays.copyOf(output.get(0), 2));
+    // verify rewritten bytes separately
+    assertThat(output.get(0)).hasSize(3);
+    assertThat(output.get(0)[2])
+        .asInstanceOf(InstanceOfAssertFactories.LONG)
+        .isGreaterThan(0L)
+        .isLessThan(dataSizeBefore);
```

Comment:
I think you can just check the Snapshot summary? Would that give us the right value?

I'm also wondering if we need this AssertThat syntax, can we just do Assert.assertEquals? Or are there benefits i'm missing.

Reply:
you're right, we can actually check the snapshot summary. I've adjusted this test to do that.
Regarding AssertJ assertions: It just provides much better context overall when an assertion fails and also allows checking for things in a more fluent way.